contracts/forecall-service/AxelarForecallService.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+import { IAxelarForecallable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IAxelarForecallable.sol';
+import { Upgradable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/upgradables/Upgradable.sol';
+import { IAxelarGateway } from '../interfaces/IAxelarGateway.sol';
+import { IAxelarForecallService } from '../interfaces/IAxelarForecallService.sol';
+import { IERC20 } from '../interfaces/IERC20.sol';
+
+// This should be owned by the microservice that is paying for gas.
+contract AxelarForecallService is Upgradable, IAxelarForecallService {
+    IAxelarGateway public immutable gateway;
+    address public immutable forecallOperator;
+
+    constructor(address gateway_, address forecallOperator_) {
```

Comment:
shouldn't this be moved to a "setup" function to work with the proxy init?

Reply:
Those are immutable variables that can be initialized only from constructor. They don't cost us storage as they become part of the bytecode